src/compiler/checker.ts

```diff
@@ -5612,7 +5698,11 @@ namespace ts {
                 }
             }
 
+            /**
+             * See signatureAssignableTo, signatureAssignableTo
+             */
             function signatureRelatedTo(source: Signature, target: Signature, reportErrors: boolean): Ternary {
+                // TODO (drosen): De-duplicate code between related functions.

```

Comment:
It looks like it's mostly down to `isRelatedTo` vs `isSignatureAssignableTo` and `Ternary` vs `boolean`. Am I missing something?